vsts/python-nightly.yaml

```diff
@@ -0,0 +1,72 @@
+name: $(BuildID)_$(BuildDefinitionName)_$(SourceBranchName) 
+
+jobs:
+- job: 'Test'
+
+  strategy:
+    maxParallel: 4
+    matrix:
+      py27_windows_mqtt:
+        pv: '2.7'
+        transport: 'mqtt'
+        imageName: 'windows-latest'
+      py27_windows_mqttws:
+        pv: '2.7'
+        transport: 'mqttws'
+        imageName: 'windows-latest'
+      py39_windows_mqtt:
+        pv: '3.9'
+        transport: 'mqtt'
+        imageName: 'windows-latest'
+      py39_windows_mqttws:
+        pv: '3.9'
+        transport: 'mqttws'
+        imageName: 'windows-latest'
+
+      py36_linux_mqtt:
+        pv: '3.6'
+        transport: 'mqtt'
+        imageName: 'Ubuntu 20.04'
+      py37_linux_mqttws:
+        pv: '3.7'
+        transport: 'mqttws'
+        imageName: 'Ubuntu 20.04'
+      py38_linux_mqtt:
+        pv: '3.8'
+        transport: 'mqtt'
+        imageName: 'Ubuntu 20.04'
+      py39_linux_mqttws:
+        pv: '3.9'
+        transport: 'mqttws'
+        imageName: 'Ubuntu 20.04'
+      py310_linux_mqtt:
+        pv: '3.10'
+        transport: 'mqtt'
+        imageName: 'Ubuntu 20.04'
+
+  pool:
+    vmImage: $(imageName)
+
+  steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: $(pv)
+        architecture: 'x64'
+
+    - script: 'python env_setup.py --no_dev'
+      displayName: 'Prepare environment (install packages + dev dependencies + test dependencies + tools)'
+
+    - script: |
+        cd $(Build.SourcesDirectory)/device_e2e
+        # "not x" means "include all tests that don't have the tag named 'x'", which is everything.
+        pytest --transport=$(transport) --junitxml=junit/TEST-python-e2e.xml -o junit_suite_name="$(Agent.JobName)" -m "not x"
+      displayName: 'E2E Device Client MQTT Connection String'
+      env:
+        IOTHUB_E2E_IOTHUB_CONNECTION_STRING: $(IOTHUB-E2E-CONNECTION-STRING)
+        IOTHUB_E2E_EVENTHUB_CONNECTION_STRING: $(IOTHUB-E2E-EVENTHUB-CONNECTION-STRING)
```

Comment:
That's sad.  Is there an event hub sdk?  It doesn't work with an Iot_hub_connection string.

Reply:
There is an eventhub sdk.  They recently did a major revision (or a rewrite) and didn't include the code to connect to eventhub using an iothub connection string.  I don't know why.